.ci/Jenkinsfile

```diff
@@ -102,20 +110,65 @@ pipeline {
                 unsuccessful {
                   archiveArtifacts(allowEmptyArchive: true, artifacts: "${MSBUILDDEBUGPATH}/**/MSBuild_*.failure.txt")
                 }
+                always {
+                  dir("${BASE_DIR}"){
+                    junit(allowEmptyResults: true, keepLongStdio: true, testResults: 'build/output/junit-*.xml')
+                    archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/output/**/*')
+                  }
+                }
               }
             }
-            stage('Deploy') {
-              when {
-                branch 'master'
-              }
-              steps {
-                dir("${BASE_DIR}"){
-                  release('secret/apm-team/ci/elastic-observability-appveyor')
-                }
+          }
+        }
+      }
+    }
+    stage('Release') {
+      when {
+        beforeAgent true
+        anyOf {
+          tag pattern: '\\d+\\.\\d+\\.\\d+(-(alpha|beta|rc)\\d*)?', comparator: 'REGEXP'
+          branch 'master'
```

Comment:
This surprises me. If this is checking against the master branch, should this be `allOf` instead of `anyOf` or is it intended to create a release on every pipeline invocation of the master branch?

Reply:
`allOf` is not possible, it is the master branch OR a tag

Reply:
A bit confusing indeed, the `Release` stage is a meta release to orchestrate the nested stages which contain their own when conditions.

- Generate artifacts locally for master and tag branches.
- Release to feedz.io only if master.
- Release to NuGet if tag
- Release Notes if tag.

I tried to describe the workflow in the description of this PR< please let me know if it's not clear to change it

Reply:
I see. This makes sense now. Thanks for the explanation.